cenpy/legacy_tiger.py

```diff
@@ -0,0 +1,229 @@
+
+        feat = link_layer.GetNextFeature()
+        geom_id_field = feat.GetFieldIndex('TLID')
+        tile_ref_field = feat.GetFieldIndex('MODULE')
+        lpoly_field = feat.GetFieldIndex('POLYIDL')
+        rpoly_field = feat.GetFieldIndex('POLYIDR')
+
+        link_count = 0
+
+        while feat is not None:
+            module = modules_hash[feat.GetField(tile_ref_field)]
+
+            tlid = feat.GetField(geom_id_field)
+
+            lpoly_id = feat.GetField(lpoly_field)
+            rpoly_id = feat.GetField(rpoly_field)
+
+            if lpoly_id == rpoly_id:
+                feat.Destroy()
+                feat = link_layer.GetNextFeature()
+                continue
+
+            try:
+                module.poly_line_links[lpoly_id].append(tlid)
+            except:
+                module.poly_line_links[lpoly_id] = [tlid]
+
+            try:
+                module.poly_line_links[rpoly_id].append(tlid)
+            except:
+                module.poly_line_links[rpoly_id] = [tlid]
+
+            link_count = link_count + 1
+
+            feat.Destroy()
+
+            feat = link_layer.GetNextFeature()
+
+        # Process all polygon features.
+
+        feat = poly_layer.GetNextFeature()
+        tile_ref_field = feat.GetFieldIndex('MODULE')
+        polyid_field = feat.GetFieldIndex('POLYID')
+
+        poly_count = 0
+        degenerate_count = 0
+
+        while feat is not None:
+            module = modules_hash[feat.GetField(tile_ref_field)]
+            polyid = feat.GetField(polyid_field)
+
+            tlid_list = module.poly_line_links[polyid]
+
+            link_coll = ogr.Geometry(type=ogr.wkbGeometryCollection)
+            for tlid in tlid_list:
+                geom = module.lines[tlid]
+                link_coll.AddGeometry(geom)
+
+            try:
+                poly = ogr.BuildPolygonFromEdges(link_coll)
+
+                if poly.GetGeometryRef(0).GetPointCount() < 4:
+                    degenerate_count = degenerate_count + 1
+                    poly.Destroy()
+                    feat.Destroy()
+                    feat = poly_layer.GetNextFeature()
+                    continue
+
+                # print poly.ExportToWkt()
+                # feat.SetGeometryDirectly( poly )
+
+                feat2 = ogr.Feature(feature_def=shp_layer.GetLayerDefn())
+
+                for fld_index in range(poly_field_count):
+                    feat2.SetField(fld_index, feat.GetField(fld_index))
+
+                feat2.SetGeometryDirectly(poly)
+
+                shp_layer.CreateFeature(feat2)
+                feat2.Destroy()
+
+                poly_count = poly_count + 1
+            except:
+                warn('BuildPolygonFromEdges failed.')
+
+            feat.Destroy()
+
+            feat = poly_layer.GetNextFeature()
+
+        if degenerate_count:
+            warn('Discarded %d degenerate polygons.' % degenerate_count)
+
+        print('Built %d polygons.' % poly_count)
```

Comment:
print should be removed. We can use `logging` if this is necessary, but I'm fine with not printing/logging here.